code/game/objects/items.dm

```diff
@@ -170,6 +173,9 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 /obj/item/Destroy()
 	flags &= ~DROPDEL	//prevent reqdels
 	QDEL_NULL(hidden_uplink)
+	if(objective_item)
+		message_admins("[src] has been destroyed at ([x],[y],[z]) in the location [loc].")
```

Comment:
might be a good idea to also have the objective item destruction loc also display it's location, as obj in obj things would be weird.